src/utilities/fastify/plugins/fastifyActiveEndpoint.spec.ts

```diff
@@ -1,75 +1,76 @@
 import fastify from 'fastify';
 
-import { getPromiseRejection, mockSpy } from '../../../testUtils/jest.js';
+import { mockSpy } from '../../../testUtils/jest.js';
 import { configureMockEnvVars } from '../../../testUtils/envVars.js';
 import type { InternetEndpoint } from '../../awala/InternetEndpoint.js';
 import { MONGODB_URI } from '../../../testUtils/db.js';
+import { InternetEndpointManager } from '../../awala/InternetEndpointManager.js';
 
 import fastifyMongoose from './fastifyMongoose.js';
-import { InternetEndpointManager } from '../../awala/InternetEndpointManager';
-import fastifyActiveEndpoint from './fastifyActiveEndpoint';
+import fastifyActiveEndpoint from './fastifyActiveEndpoint.js';
 
 const mockGetActiveEndpoint = mockSpy(jest.fn<() => Promise<InternetEndpoint>>());
 const mockMakeInitialSessionKey = mockSpy(jest.fn<() => Promise<void>>());
-// jest.unstable_mockModule('../../awala/InternetEndpointManager.js', () => ({
-//   // eslint-disable-next-line @typescript-eslint/naming-convention
-//   InternetEndpointManager: {
-//     init: () => ({
-//       getActiveEndpoint: mockGetActiveEndpoint,
-//     }),
-//   },
-// }));
-
-
 const mockEndpointManagerInit = mockSpy(jest.spyOn(InternetEndpointManager, 'init'));
 
 describe('fastifyActiveEndpoint', () => {
   configureMockEnvVars({
     MONGODB_URI,
   });
 
-  let activeEndpoint: Partial<InternetEndpoint>;
+  let mockActiveEndpoint: InternetEndpoint;
   beforeEach(() => {
-    mockEndpointManagerInit.mockResolvedValueOnce({
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+    const endpointManager: InternetEndpointManager = {
       getActiveEndpoint: mockGetActiveEndpoint,
-    } as any)
-    activeEndpoint = {
+    } as Partial<InternetEndpointManager> as any;
+
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+    mockActiveEndpoint = {
       makeInitialSessionKeyIfMissing: mockMakeInitialSessionKey,
-    };
-    mockGetActiveEndpoint.mockResolvedValueOnce(activeEndpoint as any);
+    } as Partial<InternetEndpoint> as any;
+
+    mockEndpointManagerInit.mockResolvedValueOnce(endpointManager);
+    mockGetActiveEndpoint.mockResolvedValueOnce(mockActiveEndpoint);
   });
 
-  test('Fastify should be decorated with active endpoint', async () => {
+  test('Fastify should be decorated with get active endpoint function', async () => {
     const app = fastify();
     await app.register(fastifyMongoose);
 
     await app.register(fastifyActiveEndpoint);
     await app.ready();
 
-    expect(app).toHaveProperty('activeEndpoint');
-    expect(app.activeEndpoint).toStrictEqual(activeEndpoint);
+    expect(app).toHaveProperty('getActiveEndpoint');
+    const activeEndpoint = await app.getActiveEndpoint();
+    expect(activeEndpoint).toBe(mockActiveEndpoint);
   });
 
-  test('Should make initial session key', async () => {
+  test('Active endpoint should be cached after the first request', async () => {
```

Comment:
I think this isn't needed anymore, based on the final verson of the code. It would've been necessary with the previous approach.